packages/edit-site/src/components/sidebar/index.js

```diff
@@ -24,23 +25,39 @@ const { Slot: InspectorSlot, Fill: InspectorFill } = createSlotFill(
 export const SidebarInspectorFill = InspectorFill;
 
 export function SidebarComplementaryAreaFills() {
-	const { sidebarName } = useSelect( ( select ) => {
-		let sidebar = select( interfaceStore ).getActiveComplementaryArea(
-			STORE_NAME
-		);
-
-		if ( ! [ SIDEBAR_BLOCK, SIDEBAR_TEMPLATE ].includes( sidebar ) ) {
-			sidebar = SIDEBAR_TEMPLATE;
-			if ( select( blockEditorStore ).getBlockSelectionStart() ) {
-				sidebar = SIDEBAR_BLOCK;
-			}
+	const { sidebar, isEditorSidebarOpened, hasBlockSelection } = useSelect(
+		( select ) => {
+			const _sidebar = select(
+				interfaceStore
+			).getActiveComplementaryArea( STORE_NAME );
+			const _isEditorSidebarOpened = [
+				SIDEBAR_BLOCK,
+				SIDEBAR_TEMPLATE,
+			].includes( _sidebar );
+			return {
+				sidebar: _sidebar,
+				isEditorSidebarOpened: _isEditorSidebarOpened,
+				hasBlockSelection: !! select(
+					blockEditorStore
+				).getBlockSelectionStart(),
+			};
+		},
+		[]
+	);
+	const { enableComplementaryArea } = useDispatch( interfaceStore );
+	useEffect( () => {
+		if ( ! isEditorSidebarOpened ) return;
+		if ( hasBlockSelection ) {
+			enableComplementaryArea( STORE_NAME, SIDEBAR_BLOCK );
+		} else {
+			enableComplementaryArea( STORE_NAME, SIDEBAR_TEMPLATE );
 		}
-
-		return {
-			sidebarName: sidebar,
-		};
-	} );
-
+	}, [ hasBlockSelection, isEditorSidebarOpened ] );
+	// TODO check why this is needed (when we have GS sidebar selected)..
+	let sidebarName = sidebar;
+	if ( ! isEditorSidebarOpened ) {
```

Comment:
Didn't figure out yet why this logic was needed.. Any input @david-szabo97 , @nosolosw , @youknowriad ?

Reply:
It's because `sidebarName` can become something else other than the two allowed values: SIDEBAR_BLOCK and SIDEBAR_TEMPLATE. When you open global styles panel then the `sidebarName` becomes `edit-site/global-styles`. So it will show both `DefaultSidebar` and `GlobalStylesSidebar`.

That's why we used `sidebarName` from the `useSelect` rather than directly using the one from `getActiveComplementaryArea`. You can easily fix this in your implementation by changing line 64 from:
```js
identifier={ sidebarName }
```

to 

```js
identifier={ isEditorSidebarOpened ? sidebarName : SIDEBAR_TEMPLATE }
```

Reply:
>So it will show both DefaultSidebar and GlobalStylesSidebar.

That was what confused me. That it wasn't showing both, but didn't work well too. Thanks for explaining David.

It's working as is, I didn't change that code's logic. So I think it's okay.